packages/components/src/buttons.ts

```diff
@@ -20,7 +20,7 @@ export interface ButtonPrimaryProps extends ClickableProps {
   danger?: boolean;
 }
 
-export const ButtonPrimary = (props: ButtonPrimaryProps): ReactNode => {
+export const ButtonPrimary = (props: ButtonPrimaryProps): ReactElement => {
```

Comment:
Once we upgrade to TypeScript 5.1+, ReactNode should work here.

https://www.totaltypescript.com/jsx-element-vs-react-reactnode
https://www.typescriptlang.org/docs/handbook/release-notes/typescript-5-1.html#decoupled-type-checking-between-jsx-elements-and-jsx-tag-types

Reply:
Opened #4448 to upgrade TypeScript.

Reply:
Now that TS 5.1 has merged, I'd request changing these back to ReactNode.